qa/tasks/cephfs/test_failover.py

```diff
@@ -107,12 +113,21 @@ def test_join_fs_vanilla(self):
         fs2 = self.mds_cluster.newfs(name="cephfs2")
         status, target = self._verify_init()
         active = self.fs.get_active_names(status=status)[0]
+        status2, _ = self._verify_init(fs=fs2)
+        active2 = fs2.get_active_names(status=status2)[0]
         standbys = [info['name'] for info in status.get_standbys()]
         victim = standbys.pop()
         # Set a bogus fs on the others
         for mds in standbys:
             self.config_set('mds.'+mds, 'mds_join_fs', 'cephfs2')
             self._change_target_state(target, mds, {'join_fscid': fs2.id})
+        # The active MDS for cephfs2 will be replaced by the MDS for which
+        # file system affinity has been set. Also, set the affinity for
+        # the earlier active MDS so that its is not chosen by the monitors
+        # as an active MDS for the existing file system.
+        log.info(f'assigning affinity to cephfs2 for active mds (mds.{active2})')
+        self.config_set(f'mds.{active2}', 'mds_join_fs', 'cephfs2')
```

Comment:
`self.fs.rank_fail()` at line 131 will fail `mds.cephfs` (default fs being `cephfs`), so shouldn't we make `mds.{active}` join `cephfs2`? We aren't touching fs2's rank at all.

Reply:
In that case, the monitor can choose b/w standby and active2 (which just got replaced by a standby MDS it was marked as mds_join_fs).

Reply:
standbys are already given to `cephfs` in the loop above, and `cephfs`'s mds is already marked failed, so binding `mds.cephfs2` to `cephfs2` makes sense. Although this is affinity but not guarantee, can we be always sure it would pick it up?

Reply:
If a vanilla standby is available, then the monitor will pick that up unless there is a standby with an affinity to _this_ file system.